### PR TITLE
feat(pdo): add removeEntry method to clear expired metadata

### PR DESCRIPTION
**Pull request by tamil-rss**
#1822

### CODE REVIEW

File: src/SimpleSAML/Metadata/MetaDataStorageHandlerPdo.php [hunk 1]
```diff
@@ -255,6 +255,43 @@ public function addEntry(string $index, string $set, array $entityData): bool
...
 
+    /**
+     * Remove metadata from the configured database
+     *
+     * @author - Tamilselvan Sekar, University of California, Davis. <tsekar@ucdavis.edu>
+     * 
+     * @param string $entityId The entityId we are removing up.
+     * @param string $set The set to add the metadata to.
+     *
+     * @return bool True/False if entry was successfully added
```
Comment (thijskh) [inline src/SimpleSAML/Metadata/MetaDataStorageHandlerPdo.php:266]:
```suggestion
     * @param string $entityId The entityId we are removing
     * @param string $set The set to remove the metadata from
     *
     * @return bool True/False if entry was successfully deleted
```

File: src/SimpleSAML/Metadata/MetaDataStorageHandlerPdo.php [hunk 2]
```diff
@@ -255,6 +255,43 @@ public function addEntry(string $index, string $set, array $entityData): bool
...
+
+        if ($retrivedEntityIDs !== false && count($retrivedEntityIDs) > 0) {
+            $rows = $this->db->write(
+                "DELETE FROM $tableName WHERE entity_id = :entity_id",
+                ['entity_id' => $entityId]
+            );
+            return $rows === 1;
+        }
+
+        return false;
```
[the remark this answers is not in the document]
Comment (tamil-rss) [reply src/SimpleSAML/Metadata/MetaDataStorageHandlerPdo.php:291]: I would prefer to go with `Strict`. Based on the current context, we may not consider the return value in the metarefresh module. But may be in future, the boolean value might be used in any part of the implementation considering if entity present before deleting. Just want to stay consistent as same as the **addEntry** method.

File: src/SimpleSAML/Metadata/MetaDataStorageHandlerPdo.php [hunk 3]
```diff
@@ -255,6 +255,41 @@ public function addEntry(string $index, string $set, array $entityData): bool
...
+        }
+
+        $tableName = $this->getTableName($set);
+
+        $metadata = $this->db->read(
+            "SELECT entity_id, entity_data FROM $tableName WHERE entity_id = :entity_id",
+            ['entity_id' => $entityId]
+        );
+
+        $retrivedEntityIDs = $metadata->fetch();
```
Comment (tvdijen) [inline src/SimpleSAML/Metadata/MetaDataStorageHandlerPdo.php:279]: The word is `retrieved`

File: src/SimpleSAML/Metadata/MetaDataStorageHandlerPdo.php [hunk 4]
```diff
@@ -255,6 +255,43 @@ public function addEntry(string $index, string $set, array $entityData): bool
...
+        if (!in_array($set, $this->supportedSets, true)) {
+            return false;
+        }
+
+        $tableName = $this->getTableName($set);
+
+        $metadata = $this->db->read(
+            "SELECT entity_id, entity_data FROM $tableName WHERE entity_id = :entity_id",
+            ['entity_id' => $entityId]
+        );
```
[the remark this answers is not in the document]
Comment (thijskh) [reply src/SimpleSAML/Metadata/MetaDataStorageHandlerPdo.php:277]: Why is this resolved? I think the SELECT and if statement around lines 282-286 below can just be removed? Just try the delete, if affected rows = 1 then return true else return false.

File: src/SimpleSAML/Metadata/MetaDataStorageHandlerPdo.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (thijskh) [reply src/SimpleSAML/Metadata/MetaDataStorageHandlerPdo.php:289]: I'm fine with your approach

File: src/SimpleSAML/Metadata/MetaDataStorageHandlerPdo.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (tamil-rss) [reply src/SimpleSAML/Metadata/MetaDataStorageHandlerPdo.php:277]: Sorry! missed to update the PR. Thank you

**Pull request merged**